File: src/main/java/render/GuiScreen.java
package render;

import org.joml.Matrix4f;

import java.util.ArrayList;
import java.util.List;

public abstract class GuiScreen extends GuiBase {
    private static final List<GuiBase> children = new ArrayList<>();

    protected void appendChild(GuiBase c) {
        children.add(c);
    }

    public void mouseMovement(float x, float y) {
        for (GuiBase c : children) {
            c.mouseMovement(x,y);
        }
    }

    public void mouseClick(int button, int action) {
        for (GuiBase c : children) {
            c.mouseClick(button, action);
        }
    }

    public void render(Matrix4f proj) {
        for (GuiBase c : children) {
            c.render(proj);
        }
    }
}

File: src/main/kotlin/render/GuiScreen.kt
package render

import org.joml.Matrix4f

abstract class GuiScreen : GuiBase() {
    val children: MutableList<GuiBase> = mutableListOf()

    protected fun appendChild(c: GuiBase) {
        children.add(c)
    }

    override fun mouseMovement(x: Float, y: Float) {
        for (c in children) {
            c.mouseMovement(x,y)
        }
    }

    override fun mouseClick(button: Int, action: Int) {
        for (c in children) {
            c.mouseClick(button, action)
        }
    }

    override fun render(proj: Matrix4f) {
        for (c in children) {
            c.render(proj)
        }
    }
}